homeassistant/components/dynalite/bridge.py

```diff
@@ -31,7 +31,7 @@ def __init__(self, hass, config):
         LOGGER.debug("Setting up bridge - host %s", self.host)
         return await self.dynalite_devices.async_setup()
 
-    async def reload_config(self, config):
+    def reload_config(self, config):
```

Comment:
We're still awaiting this function in the entry update listener.

Reply:
I know. the reload_config is a sync function, and the library is built in a way that reload_config can be called at any time, as long as it is called from the same thread of the event loop.
However, I didn't like this too much either, as it is confusing, but couldn't find another solution, so definitely open to any suggestions. My problem is that if the entry is already defined, but the config has changed, I need to call the update. However, the bridge is not yet fully set up at this point as this happens in parallel. If there is a way to wait until the config entry is set up before loading the new config from configuration.yaml, then I would only call it at that time. However, all the solutions I could think off were even uglier, so would appreciate any insight on another component that is doing it right and I could use as reference.
Note that it does work and I couldnt find any bugs in the flow, but it is not the best design pattern and will happily change it if I can find a cleaner solution

Reply:
I don't think this can work. It's not possible to await non awaitable function.

Reply:
Why not? reload_config does not await on anything. It only updates internal data structures, which are designed that they could change during any await in the async_setup (or others), so it does not depend on async_setup being completed. 
That said, I like your suggestion below, which will make it possible to write a clean code that does not require explanations (and therefore also safer). Will give it a shot tonight

Reply:
We're still awaiting this function, that's the problem.

Reply:
https://github.com/home-assistant/core/blob/007d934214f5a92db69ac3fbfe5470a54d19a459/homeassistant/components/dynalite/__init__.py#L144

Reply:
you are correct. my mistake. fixed in the PR I am preparing

Reply:
just found out why it wasn't caught in the tests. it still behaved "ok" since it evaluated "bridge.reload_config(entry.data)" as it is a sync function. then it tried to await its return value, None, and threw an assertion. However, since this is the last meaningful statement (except a log), the test still showed that everything happened.
BTW, I believe the code would also behave well in that case except for the log about the exception, but obviously i fixed it to remove the await.
I was wondering if there is a way to make pytest fail in case there are uncaught exception, not awaited coroutines, etc. to avoid these instances and verify a cleaner code

Reply:
Non awaited coroutines should show up in the warning log when running pytest I think.

Reply:
But here it's the other way, right? We're awaiting when we shouldn't. Not sure about that one.

Reply:
that is correct. it is not a good pattern anyway and i fixed it. It throws an exception. The problem is that the test succeeds. I would like a way to fail it on assertions / non awaited, etc. as i shouldn't have them in the code
BTW, it can be a bit tricky since i get many tests with an uncaught exception for persistent_notification/create:
ERROR    homeassistant.core:core.py:143 Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/home/ziv/switch/home-assistant/homeassistant/core.py", line 1209, in async_call
    raise ServiceNotFound(domain, service) from None
homeassistant.exceptions.ServiceNotFound: Unable to find service persistent_notification/dismiss

I am not doing anything directly with persistent_notification, but it does seem to me that tests should complete with no warnings at all